### ISSUE TITLE
Support arbitrary OCI annotations

### ISSUE THREAD

**Issue by Vogtinator** [SUSE/kiwi#851]
Like docker labels, OCI annotations should be supported as well.

See https://github.com/opencontainers/image-spec/blob/master/annotations.md for examples.

umoci has full support for those.

**Comment by davidcassany** [comment]
@fvogt while I agree it would be nice to support OCI annotations I don't think we should implement it right now. Mostly because I am not aware of anyone using OCI image builds in KIWI (`<type image="oci">`).  OCI annotations are lost when converting to docker format as far as I am aware.

Probably @vrothberg can give us some further details about how skopeo does the OCI to Docker conversion. Are OCI annotations kept or taken into account in some way?

I think it is better to stick to labels until there is a clear usage of this annotations. It could happen they are never used and it means adding extra tags in kiwi schema and some additional black magic to add some predefined metadata.

**Comment by Vogtinator** [comment]
> Mostly because I am not aware of anyone using OCI image builds in KIWI (<type image="oci">). OCI annotations are lost when converting to docker format as far as I am aware.

The goal is to use OCI annotations together with docker labels in the registry. If the annotations are lost when converting from docker format, I can just switch to `<type image="oci"/>`.

**Comment by vrothberg** [comment]
> Probably @vrothberg can give us some further details about how skopeo does the OCI to Docker conversion. Are OCI annotations kept or taken into account in some way?

That's an interesting question. I'd need to test it. Maybe, we can discuss this all together next week?

**Comment by schaefi** [comment]
yes good point, we should take the chance for a get together :+1: I'll be in NUE, Wed - Fri

**Comment by davidcassany** [comment]
> yes good point, we should take the chance for a get together +1 I'll be in NUE, Wed - Fri

Great! Then yes, this is something to discuss.

**Comment by cyphar** [comment]
I can also pop in for this discussion.

Regarding arbitrary annotations, umoci does have support for them internally but setting them for individual descriptors is currently a bit of a UX problem -- I haven't figured out a reasonable way of having a CLI interface to let to manipulate individual descriptor pointers.

**Comment by Vogtinator** [comment]
Well, it needs to be done in the tools first and then the infrastructure can be built around it.

I can't reopen this issue myself.
